client/my-sites/checkout/checkout-thank-you/utils.ts

```diff
@@ -104,3 +105,22 @@ export function isDomainOnly( purchases: ReceiptPurchase[] ): boolean {
 		)
 	);
 }
+
+export type FindPredicate = (
+	product: ( WithSnakeCaseSlug | WithCamelCaseSlug ) & {
+		is_domain_registration?: boolean;
+		isDomainRegistration?: boolean;
+		meta: string;
+	}
+) => boolean;
+
+export function getDomainPurchaseType( purchases: ReceiptPurchase[] ): [ string, FindPredicate ] {
+	const hasDomainMapping = purchases.some( isDomainMapping );
+
+	if ( hasDomainMapping && purchases.some( isDomainRegistration ) ) {
+		return [ 'REGISTRATION', isDomainRegistration ];
+	} else if ( hasDomainMapping ) {
+		return [ 'MAPPING', isDomainMapping ];
+	}
+	return [ 'TRANSFER', isDomainTransfer ];
```

Comment:
Small suggestion to make that block of code easier to read:
```suggestion
	if ( hasDomainMapping && purchases.some( isDomainRegistration ) ) {
		return [ 'REGISTRATION', isDomainRegistration ];
	}
	
	if ( hasDomainMapping ) {
		return [ 'MAPPING', isDomainMapping ];
	}
	
	return [ 'TRANSFER', isDomainTransfer ];
```